core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -2084,7 +2087,8 @@ public boolean processWalBlock(
                         // Can commit without O3 and LAG has just enough rows
                         || (commitToTimestamp >= newMaxLagTimestamp && totalUncommitted > getMetaMaxUncommittedRows())
                         // Too many uncommitted transactions in LAG
-                        || (configuration.getWalMaxLagTxnCount() > 0 && txWriter.getLagTxnCount() >= configuration.getWalMaxLagTxnCount());
+                        || (configuration.getWalMaxLagTxnCount() > 0 && txWriter.getLagTxnCount() >= configuration.getWalMaxLagTxnCount())
+                        || (configuration.getMicrosecondClock().getTicks() - lastWalCommitTimestampMicros > configuration.getCommitLatency());
```

Comment:
Am I right in understanding that the goal here to make data visibility more predictable? For larger transactions this change doesn't affect the server behavior because of uncommitted rows threshold, so the patch is aimed at lots of small transactions.

Reply:
Andrei, you are correct. There should not be any impact on how we process large transactions. It wasn't the aim anyway

Reply:
Great, thanks for the confirmation. Then, apart from minor comments, the PR looks good to me.